mlrun/datastore/targets.py

```diff
@@ -1115,6 +1115,8 @@ def write_dataframe(
             df = self.prepare_spark_df(df)
             df.write.mode("overwrite").save(**options)
         else:
+            # To prevent modification of the original dataframe
+            df = df.copy(deep=False)
```

Comment:
Is this actually a problem? It doesn't look like `frames_client.write` modifies the df passed to it.

Reply:
@gtopper you can run and see it is modifies the df

Reply:
Opened https://jira.iguazeng.com/browse/IG-21310 for this issue.